test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java

```diff
@@ -551,6 +566,232 @@ public void narrativeDecoration() throws Exception {
+                        "{\"whee\": \"imaprettypony1\"}",
+                        ImmutableMap.of("whee", Arrays.asList("imaprettypony1")))),
+                false);  // isPublic set to false
+
+        indexStorage.indexObjects(
+                ObjectTypeParsingRules.getBuilder(
+                        new SearchObjectType("Deco", 1),
+                        new StorageObjectType("foo", "bar"))
+                        .withIndexingRule(IndexingRules.fromPath(new ObjectJsonPath("whee"))
+                                .build())
+                        .build(),
+                SourceData.getBuilder(new UObject(new HashMap<>()), "objname2", "creator")
+                        .build(),
+                Instant.ofEpochMilli(10000),
+                null,
+                new GUID("WS:2/1/1"),
+                ImmutableMap.of(new GUID("WS:2/1/1"), new ParsedObject(
+                        "{\"whee\": \"imaprettypony1\"}",
+                        ImmutableMap.of("whee", Arrays.asList("imaprettypony1")))),
+                true);  // isPublic set to true
+    }
+
+    private ObjectData setUpOptionalAuthForSearchGetRes(String guid, String objName) throws Exception{
+        return new ObjectData()
+                .withData(new UObject(ImmutableMap.of("whee", "imaprettypony1")))
+                .withGuid(guid)
+                .withKeyProps(ImmutableMap.of("whee", "imaprettypony1"))
+                .withCreator("creator")
+                .withType("Deco")
+                .withTypeVer(1L)
+                .withObjectName(objName)
+                .withTimestamp(10000L)
+                .withHighlight(new HashMap<>());
+    }
+    @Test
+    public void optionalAuthSearchObjects() throws Exception{
+        setUpOptionalAuthForSearch();
+
+        final ObjectData expected1 = setUpOptionalAuthForSearchGetRes("WS:1/1/1","objname1");
+        final ObjectData expected2 = setUpOptionalAuthForSearchGetRes("WS:2/1/1","objname2");
+
+        SearchObjectsOutput authSearchObjsOut;
+        SearchObjectsOutput noAuthSearchObjsOut;
+
+
+        try {
+            noAuthSearchObjsOut =  noAuthSearchCli.searchObjects(new SearchObjectsInput()
+                    .withAccessFilter(new AccessFilter())
+                    .withMatchFilter(new MatchFilter()));
+
+            authSearchObjsOut =  searchCli.searchObjects(new SearchObjectsInput()
+                    .withAccessFilter(new AccessFilter())
+                    .withMatchFilter(new MatchFilter()));
+        } catch (ServerException e) {
+            System.out.println("Exception server side trace:\n" + e.getData());
+            throw e;
+        }
+
+        assertThat("incorrect search objects count", noAuthSearchObjsOut.getObjects().size(), is(1));
+        TestCommon.compare(noAuthSearchObjsOut.getObjects().get(0), expected2);
+
+        assertThat("incorrect search objects count", authSearchObjsOut.getObjects().size(), is(2));
+
+        Set<String> res = new HashSet<>();
```

Comment:
Something like this is less convoluted:
```
final Set<String> got = authSearchObjectsOut.getObjects().stream()
    .map(o -> o.getGuid()).collect(Collectors.toSet);
assertThat("incorrect results", got, is(set("WS:1/1/1", "WS:2/1/1")));
```

Then you don't need the size check either.